File: collabfrontend/WebContent/Controller/JobController.js
/**
 * 
 */
app.controller('JobController',function($scope,$location,JobService){
	console.log("job Controller loading.......")
	
	$scope.showjobdetails=false
	$scope.savejob=function(){
		JobService.savejob($scope.job).then(function(response){
		$scope.success="Job entered successfully.."
			$location.path('/savejob')
	},function(response){
		console.log(response.status)
		$scope.error=response.data  
				if(response.status==401)
					$location.path('/login')
				$location.path('/savejob')
			
	})
	}
	
	$scope.applyjob=function(job){
			
		JobService.saveappliedjob(job).then(function(response){
			console.log(response.status)
			listofjobs()
			listofappliedjobs()
			$location.path('/getalljobs')
	},function(response){
		if(response.status==401)
					$location.path('/login')
	})
	}
	
	
	$scope.getjobbyid=function(id){
		JobService.getjobbyid(id).then(function(response){
			$scope.showjobdetails=true
			$scope.job=response.data
	},function(response){
		console.log(response.status)
		$scope.error=response.data  
				if($scope.error.code==5)
					$scope.unauthorized=response.data
				$location.path('/login')
	})
	}
	
	function listofjobs(){
		JobService.getalljobs().then(function(response){
			console.log(response.status)
			$scope.jobs=response.data
			isapplied($scope.jobs)
	},function(response){
		console.log(response.status)
		$scope.error=response.data  
				if($scope.error.code==5)
					$scope.unauthorized=response.data
				$location.path('/login')
	})
	}
	
	function listofappliedjobs(){
		JobService.getallappliedjobs().then(function(response){
			console.log(response.status)
			$scope.appliedjobs=response.data
	},function(response){
		console.log(response.status)
				if($scope.error.code==5)
					$location.path('/login')
	})
	}
	
	function isapplied(jobs){
		JobService.isapplied(jobs).then(function(response){
			console.log(response.status)
			$scope.applied=response.data
		},function(response){
			console.log(response.status)
			$location.path('/login')
		})
	}
	listofjobs()
	listofappliedjobs()
})

File: collabbackend/src/main/java/com/niit/controller/JobController.java
package com.niit.controller;



import java.util.Date;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import javax.servlet.http.HttpSession;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

import com.niit.dao.Jobdao;
import com.niit.dao.Userdao;
import com.niit.model.AppliedJob;
import com.niit.model.Error;
import com.niit.model.Job;
import com.niit.model.User;

@Controller
public class JobController {
	@Autowired
	private Userdao userdao;
	
	@Autowired
	private Jobdao jobdao;
	
	
	@RequestMapping(value="/saveJob", method=RequestMethod.POST)
	public ResponseEntity<?> savejob(@RequestBody Job job,HttpSession session){
		
		if( session.getAttribute("username")==null){
			Error error=new Error(5,"Unauthorized access");
			return new ResponseEntity<Error>(error,HttpStatus.UNAUTHORIZED);
		}
		String username=(String) session.getAttribute("username");
		User user=userdao.getUserbyUsername(username);
		if(user.getRole().equals("ADMIN")){
			try{
				Date date=new Date();
				job.setPostedOn(date);
				jobdao.saveJob(job);
				return new ResponseEntity<Job>(job,HttpStatus.OK);
			}
			catch(Exception e){
				Error error=new Error(7,"Unable to enter job details");
				return new ResponseEntity<Error>(error,HttpStatus.INTERNAL_SERVER_ERROR);
			}
		}
		else{
			Error error=new Error(6,"Access denied");
			return new ResponseEntity<Error>(error,HttpStatus.UNAUTHORIZED);
			
		}
	}
	
	@RequestMapping(value="/getAlljobs", method=RequestMethod.GET)
	public ResponseEntity<?> getalljobs(HttpSession session){
		if(session.getAttribute("username")==null){
			Error error=new Error(5,"Unauthorized access");
			return new ResponseEntity<Error>(error,HttpStatus.UNAUTHORIZED);
		}
		List<Job> jobs=jobdao.getalljobs();
		return new ResponseEntity<List<Job>>(jobs,HttpStatus.OK);
		
	}

	@RequestMapping(value="/getjobByid/{id}", method=RequestMethod.GET)
	public ResponseEntity<?> getjobbyid(HttpSession session,@PathVariable int id){
		if(session.getAttribute("username")==null){
			Error error=new Error(5,"Unauthorized access");
			return new ResponseEntity<Error>(error,HttpStatus.UNAUTHORIZED);
		}
		Job job=jobdao.getjobbyid(id);
		return new ResponseEntity<Job>(job,HttpStatus.OK);
		
	}
	
	@RequestMapping(value="/applyJob", method=RequestMethod.POST)
	public ResponseEntity<?> applyjob(@RequestBody Job job,HttpSession session){
		
		if( session.getAttribute("username")==null){
			Error error=new Error(5,"Unauthorized access");
			return new ResponseEntity<Error>(error,HttpStatus.UNAUTHORIZED);
		}
		String username=(String) session.getAttribute("username");
		User user=userdao.getUserbyUsername(username);
		AppliedJob applyingjob= new AppliedJob();
		applyingjob.setUsers(user);
		applyingjob.setJobs(job);
		jobdao.applyJob(applyingjob);
		return new ResponseEntity<AppliedJob>(applyingjob,HttpStatus.OK);
	}
	
	@RequestMapping(value="/getAllappliedjobs", method=RequestMethod.GET)
	public ResponseEntity<?> getallappliedjobs(HttpSession session){
		if(session.getAttribute("username")==null){
			Error error=new Error(5,"Unauthorized access");
			return new ResponseEntity<Error>(error,HttpStatus.UNAUTHORIZED);
		}
		List<AppliedJob> AppliedJobs=jobdao.getappliedjobs();
		return new ResponseEntity<List<AppliedJob>>(AppliedJobs,HttpStatus.OK);
		
	}
	
	@RequestMapping(value="/getappliedJobs", method=RequestMethod.GET)
	public ResponseEntity<?> applyjob(HttpSession session){
		
		if( session.getAttribute("username")==null){
			Error error=new Error(5,"Unauthorized access");
			return new ResponseEntity<Error>(error,HttpStatus.UNAUTHORIZED);
		}
		String username=(String) session.getAttribute("username");
		List<AppliedJob> appliedjobs= jobdao.getappliedjobsbyuser(username);
		return new ResponseEntity<List<AppliedJob>>(appliedjobs,HttpStatus.OK);
	}
	 @RequestMapping(value="/isapplied",method=RequestMethod.PUT)
	    public ResponseEntity<?> getMutualFriends(@RequestBody List<Job> jobs,HttpSession session){
	    	String username=(String)session.getAttribute("username");
	    	if(username==null){
	    		Error error=new Error(5,"UnAuthorized Access..");
	    		return new ResponseEntity<Error>(error,HttpStatus.UNAUTHORIZED);
	    	}
	    	Map<String, Boolean> isapplied=new HashMap<String, Boolean>();
	    	for(Job j :jobs){
	    		isapplied.put(j.getJobtitle(), jobdao.isjobapplied(username, j.getJobId()));
	    	System.out.println(isapplied.size());
	    	}
	    	System.out.println(isapplied);
	    	return new ResponseEntity<Map<String,Boolean>>(isapplied,HttpStatus.OK);
	    }
	
}
